### PR TITLE
update fix for block 453100

### PR DESCRIPTION
**Pull request by nandofw**
-added collateral amount on CDeterministicMNState this is used to check if the collateral is valid at a determined block height
-reverted #254 as is no longer necessary

this pr require a sync from 0

### CODE REVIEW

**Review by npq7721 — APPROVED** [review]
good solution, LGTM

**Comment by nandofw** [comment]
fixed tests error when collateral is a output of the TRANSACTION_PROVIDER_REGISTER

File: src/evo/deterministicmns.h [hunk 1]
```diff
@@ -48,6 +48,8 @@ class CDeterministicMNState
     // sha256(proTxHash, confirmedHash) to speed up quorum calculations
     // please note that this is NOT a double-sha256 hash
     uint256 confirmedHashWithProRegTxHash;
+    //collateral amount used in quorum calculations
+    CAmount ncollateralamount;
```
Comment (CodeIsTheKey) [inline src/evo/deterministicmns.h]: Suggest changing case to match other fields: `nCollateralAmount`

File: src/evo/deterministicmns.h [hunk 2]
```diff
@@ -154,6 +157,7 @@ class CDeterministicMNStateDiff
         Field_addr                              = 0x0800,
         Field_scriptPayout                      = 0x1000,
         Field_scriptOperatorPayout              = 0x2000,
+        Field_ncollateralamount                 = 0x4000,
```
Comment (CodeIsTheKey) [inline src/evo/deterministicmns.h]: Suggest `Field_nCollateralAmount`

**Review by CodeIsTheKey — APPROVED** [review]
Great solution!  Nice way to avoid the index and get a good collateral check.

File: src/evo/deterministicmns.h (hunk 2 above)
Comment (nandofw) [reply src/evo/deterministicmns.h]: done

**Pull request merged**